iceoryx_utils/include/iceoryx_utils/internal/cxx/vector.inl

```diff
@@ -268,11 +268,8 @@ inline T& vector<T, Capacity>::at(const uint64_t index) noexcept
 template <typename T, uint64_t Capacity>
 inline const T& vector<T, Capacity>::at(const uint64_t index) const noexcept
 {
-    if (index + 1u > m_size)
-    {
-        std::cerr << "out of bounds access, current size is " << m_size << " but given index is " << index << std::endl;
-        std::terminate();
-    }
+    // No out of bound accesses shall be possible
+    cxx::Expects(!(index + 1u > m_size) && "Out of bounds access");
```

Comment:
Negations makes it harder to reason about complex conditions. Why not just `cxx::Expects(index < m_size && "...");`? This has the additional benefit to be more performant

Reply:
Done